glambie/processing/process_regional_results.py

```diff
@@ -343,12 +407,140 @@ def _run_region_timeseries_one_source(
+        Only used if method_to_correct_seasonally is defined as seasonal homogenization
+    annual_combined_dataset : Timeseries
+        a single timeseries of all the annual datasets combined
+        output from '_run_region_variability_for_one_source()'
+    annual_backup_dataset : Timeseries
+        Dataset to be used as a backup when to extend combined annual series when it doesn't cover the whole period
+    year_type : YearType
+        type of year to be used, e.g calendar or glaciological
+    method_to_extract_trends : ExtractTrendsMethod
+        method as to how the long-term trends are extracted from a high resolution (e.g. monthly) timeseries
+    method_to_correct_seasonally : SeasonalCorrectionMethod
+        method as to how long-term trends are correct when they don't start in the desired season, i.e. don't follow
+        the desired annual grid defined with 'year_type'    data_group : GlambieDataGroup
+        _description_
+    min_max_time_window_for_longterm_trends : Tuple[float, float], optional
+        if specified, the time series are filtered by the time window before the longterm trend is extracted,
+        meaning that the resulting longterm trends are within the minimum and maximum of the time window.
+        Note that existing longterm trends are removed if they are outside the time window.
+        The dates are expected in decimal years format (float), e.g. 2012.75.
+        by default None
+
+    Returns
+    -------
+    Tuple[Timeseries, DataCatalogue, DataCatalogue]
+        1) a single timeseries of all the trend datasets combined
+        2) a catalogue of the homogenized trends
+        3) a data catalogue of all the calibrated trends that were averaged
+    """
+
+    log.info("Recalibrating with longterm trends within data group and region...")
+    # get catalogue with longerm datasets in same unit as calibration dataset (mwe)
+    data_catalogue_trends_homogenized = convert_datasets_to_longterm_trends_in_unit_mwe(
+        data_catalogue_trends, year_type=year_type,
+        seasonal_calibration_dataset=seasonal_calibration_dataset,
+        method_to_extract_trends=method_to_extract_trends,
+        method_to_correct_seasonally=method_to_correct_seasonally,
+        output_trend_date_range=min_max_time_window_for_longterm_trends)
+
+    # now treat case where trends are outside annual combined timeseries
+    annual_combined_full_ext = extend_annual_timeseries_if_shorter_than_time_window(
+        annual_timeseries=annual_combined_dataset,
+        timeseries_for_extension=annual_backup_dataset.convert_timeseries_to_annual_trends(year_type=year_type),
+        desired_time_window=data_catalogue_trends_homogenized.get_time_span_of_datasets())
+
+    # recalibrate
+    catalogue_calibrated_series = calibrate_timeseries_with_trends_catalogue(
+        data_catalogue_trends_homogenized, annual_combined_full_ext)
+    # we dont remove trends as these are calibrated series with trends
```

Comment:
I don't understand this comment, can you re-write a bit?

Reply:
done